tactics/tacticals.ml

```diff
@@ -734,13 +732,11 @@ module New = struct
   let case_nodep_then_using =
     general_elim_then_using gl_make_case_nodep false
 
-  let pf_constr_of_global ref tac =
-    Proofview.Goal.enter { enter = begin fun gl ->
-      let env = Proofview.Goal.env gl in
-      let sigma = Tacmach.New.project gl in
-      let (sigma, c) = Evd.fresh_global env sigma ref in
-      let c = EConstr.of_constr c in
-      Proofview.Unsafe.tclEVARS sigma <*> (tac c)
-    end }
+  let pf_constr_of_global ref =
```

Comment:
The type changes, so the semantic changes indeed. The new version is more atomic as it does not perform a focus on the goal, which is delegated to the caller. This means in particular that one needs to scope the function inside a Goal.enter to generate distinct instances.